pennylane/liealg/cartan_decomp.py

```diff
@@ -0,0 +1,235 @@
+    >>> ops1 = [qml.X(0)]
+    >>> ops2 = [qml.Y(0)]
+    >>> vspace1 = [qml.X(0), qml.Y(0)]
+
+    Because :math:`[X_0, Y_0] = 2i Z_0`, the commutators do not map to the selected vector space.
+
+    >>> check_commutation(ops1, ops2, vspace1)
+    False
+
+    Instead, we need the full :math:`\mathfrak{su}(2)` space.
+
+    >>> vspace2 = [qml.X(0), qml.Y(0), qml.Z(0)]
+    >>> check_commutation(ops1, ops2, vspace2)
+    True
+    """
+
+    if any(isinstance(op, Operator) for op in ops1):
+        ops1 = [op.pauli_rep for op in ops1]
+
+    if any(isinstance(op, Operator) for op in ops2):
+        ops2 = [op.pauli_rep for op in ops2]
+
+    if not isinstance(vspace, qml.pauli.PauliVSpace):
+        vspace = qml.pauli.PauliVSpace(vspace, dtype=complex)
+
+    for o1 in ops1:
+        for o2 in ops2:
+            com = o1.commutator(o2)
+            com.simplify()
+            if len(com) != 0:
+                if vspace.is_independent(com):
+                    return False
+
+    return True
+
+
+def check_cartan_decomp(k: List[PauliSentence], m: List[PauliSentence], verbose=True):
+    r"""Helper function to check the validity of a Cartan decomposition :math:`\mathfrak{g} = \mathfrak{k} \oplus \mathfrak{m}.`
+
+    Check whether of not the following properties are fulfilled.
+
+    .. math::
+
+            [\mathfrak{k}, \mathfrak{k}] \subseteq \mathfrak{k} & \text{ (subalgebra)}\\
+            [\mathfrak{k}, \mathfrak{m}] \subseteq \mathfrak{m} & \text{ (reductive property)}\\
+            [\mathfrak{m}, \mathfrak{m}] \subseteq \mathfrak{k} & \text{ (symmetric property)}
+
+    .. warning:: This function is expensive to compute
+
+    Args:
+        k (List[PauliSentence]): List of operators of the vertical subspace
+        m (List[PauliSentence]): List of operators of the horizontal subspace
+        verbose: Whether failures to meet one of the criteria should be printed
+
+    Returns:
+        bool: Whether or not all properties are fulfilled
+
+    .. seealso:: :func:`~cartan_decomp`
+
+    **Example**
+
+    We first construct a Lie algebra.
+
+    >>> from pennylane import X, Z
+    >>> from pennylane.labs.dla import concurrence_involution, even_odd_involution, cartan_decomp
+    >>> generators = [X(0) @ X(1), Z(0), Z(1)]
+    >>> g = qml.lie_closure(generators)
+    >>> g
+    [X(0) @ X(1),
+     Z(0),
+     Z(1),
+     -1.0 * (Y(0) @ X(1)),
+     -1.0 * (X(0) @ Y(1)),
+     -1.0 * (Y(0) @ Y(1))]
+
+    We compute the Cartan decomposition with respect to the :func:`~concurrence_involution`.
+
+    >>> k, m = cartan_decomp(g, concurrence_involution)
+    >>> k, m
+    ([-1.0 * (Y(0) @ X(1)), -1.0 * (X(0) @ Y(1))],
+     [X(0) @ X(1), Z(0), Z(1), -1.0 * (Y(0) @ Y(1))])
+
+    We can check the validity of the decomposition using ``check_cartan_decomp``.
+
+    >>> from pennylane.labs.dla import check_cartan_decomp
+    >>> check_cartan_decomp(k, m)
+    True
+
+    """
+    # TODO: implement proper matrix implementation
```

Comment:
Shall we track this in SC instead? :)

Reply:
I wanted to do this in this PR anyway :) pushed a version just now, but let me check that all tests pass before reviewing